### PR TITLE
Feat/user profile mobile UI

### PR DESCRIPTION
**Pull request by Wiecek-K**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
Based on the provided task requirements and the code changes, here is my review:

1. **app/(profile)/layout.tsx**: The change from `MyProfileHeader` to `DefaultHeader` seems to be in line with the task requirements. The `DefaultHeader` appears to be more responsive and suitable for mobile devices.

2. **app/(profile)/user-profile/page.module.scss**: The added CSS rules for mobile devices seem to be in line with the task requirements. The padding and margin adjustments should help the profile UI display better on mobile devices.

3. **src/components/Headers/DefaultHeader/DefaultHeader.module.scss**: The changes here seem to be focused on making the header more responsive. The height, padding, and margin adjustments, as well as the positioning of elements, should help the header display better on mobile devices.

4. **src/components/Headers/DefaultHeader/DefaultHeader.tsx**: The changes here seem to be focused on adjusting the rendering of buttons based on the authentication status. This should not affect the mobile responsiveness of the profile UI.

5. **src/components/MyProfile/ProfileMain/ProfileMain.module.scss**: The removal of padding in the mobile view and adjustment of margin should help improve the display of the profile main section on mobile devices.

6. **src/components/MyProfile/ProfileMain/ProfileMain.tsx**: The removal of `GoBackButton` import does not seem to affect the mobile responsiveness of the profile UI.

7. **src/components/UserProfile/UserProfileDetails/UserProfileDetails.module.scss**: The added CSS rules for mobile devices, including padding, margin, and display adjustments, should help improve the display of user profile details on mobile devices.

8. **src/components/UserProfile/UserProfileMain/UserProfileMain.module.scss**: The added CSS rules for mobile devices, including positioning, padding, margin, and display adjustments, should help improve the display of the user profile main section on mobile devices.

9. **src/components/UserProfile/UserProfileMain/UserProfileMain.tsx**: The rearrangement of elements and addition of a back button should improve navigation on mobile devices.

Overall, it seems that the code changes are in line with the task requirements of adjusting the Profile UI for better display on mobile devices. However, without testing these changes on actual devices or emulators, it's hard to definitively say if all requirements have been met.

File: src/components/Headers/DefaultHeader/DefaultHeader.tsx [hunk 1]
```diff
@@ -14,46 +14,35 @@ const DefaultHeader = () => {
...
-        <Button onClick={() => signIn()} variant={'primary'}>
-          Create profile
-          <AddIcon />
-        </Button>
-      </div>
+      {status === 'authenticated' ? (
+        <div className={styles.frameButtons}>
+          <Button
+            onClick={() => router.push('/my-profile')}
+            variant={'primary'}
```
Comment (wawrzonnn) [inline src/components/Headers/DefaultHeader/DefaultHeader.tsx:27]: You should use routes from src/utils

**Review by wawrzonnn — APPROVED** [review]

**Review by hansac91 — APPROVED** [review]

File: src/components/Headers/DefaultHeader/DefaultHeader.tsx (hunk 1 above)
Comment (Sumick) [reply src/components/Headers/DefaultHeader/DefaultHeader.tsx:27]: **nitpick**: What wawrozn means is: ` onClick={() => router.push(AppRoutes.MyProfile)}`

**Review by Sumick — APPROVED** [review]
Looks good! Great job, Im leaving one non-blocking comment.

**Pull request merged**